plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/connection/ConnectionWizardPageModel.java

```diff
@@ -176,60 +205,99 @@ public void createConnection() {
 	 *            the host to get the connection for.
 	 */
 	private void createConnection(String host) {
+		if (connectionFactory == null
+				|| StringUtils.isEmpty(host)) {
+			return;
+		}
+		if (connectionFactory instanceof AutomaticConnectionFactoryMarker) {
+			createConnection(host, connectionsFactory);
+		} else {
+			createConnection(host, connectionFactory);
+		}
+	}
+
+	private void createConnection(String host, IConnectionFactory factory) {
 		IConnection connection = null;
-		if (connectionFactory != null
-				&& !StringUtils.isEmpty(host)) {
-			if (connectionFactory instanceof AutomaticConnectionFactoryMarker) {
-				try {
-					connection = connectionsFactory.create(host);
-					if (connection == null) {
-						setConnectionCreated(StatusFactory.errorStatus(
-								OpenShiftCommonUIActivator.PLUGIN_ID, NLS.bind("The host at {0} is no OpenShift host", host)));
-					} else {
-						// switches <automatic> to detected type
-						setConnectionFactory(connectionsFactory.getByConnection(connection.getClass()));
-					}
-				} catch (IOException e) {
-					resetConnection(ValidationStatus.error(NLS.bind("Could not connect to host at {0}.", host), e));
-				}
+		try {
+			connection = factory.create(host);
+			if (equalsTypeAndHost(connection, this.connection)) {
+				return;
+			}
+			
+			if (connection.canConnect()) {
+				updateModel(connection, 
+						factory, 
+						Status.OK_STATUS, 
+						Status.OK_STATUS);
 			} else {
-				try {
-					connection = connectionFactory.create(host);
-					if (!connection.canConnect()) {
-						connection = null;
-					}
-				} catch (IOException e) {
-					resetConnection(ValidationStatus.error(NLS.bind("Could not connect to host at {0}.", host), e));
-				}
+				updateModel(null, 
+						factory, 
+						ValidationStatus.error(NLS.bind("Host at {0} is not an {1} host.", host, factory.getName())), 
+						Status.OK_STATUS);
 			}
+		} catch (IOException e) {
+			updateModel(null, 
+					factory, 
+					ValidationStatus.error(NLS.bind("Could not connect to host at {0}.", host), e), 
+					Status.OK_STATUS);
 		}
+	}
 
-		setConnection(connection);
+	private void createConnection(String host, IConnectionsFactory connectionsFactory) {
+		IConnection connection;
+		try {
+			connection = connectionsFactory.create(host);
+			if (connection != null) {
+				updateModel(connection, 
+						connectionFactory, 
+						Status.OK_STATUS,
+						Status.OK_STATUS);
+			} else {
+				updateModel(null, 
+						connectionFactory,
+						ValidationStatus.error(NLS.bind("The host at {0} is no OpenShift host", host)),
+						Status.OK_STATUS);
+			}
+		} catch (IOException e) {
+			updateModel(null, 
+					AutomaticConnectionFactoryMarker.getInstance(),
+					ValidationStatus.error(NLS.bind("Could not connect to host at {0}.", host), e), 
+					Status.OK_STATUS);
+		}
+	}
+
+	private boolean equalsTypeAndHost(IConnection thisConnection, IConnection thatConnection) {
```

Comment:
@adietish this and that....anyway to distinguish them more or is this just basically an 'equals' method?